### ISSUE TITLE
Few questions regarding the attention model implementation

### ISSUE THREAD

**Issue by jennyzu** [SaoYan/LearnToPayAttention#1]
First of all, thank you very much for the code.
I have a few questions, hoping you can help.
1. Why are you using sigmoid function and not softmax?
2. Why are you using adaptive_avg_pool2d? I haven't seen it in the paper.
I guess the first two questions are related, since normalize_attn is True when running the code. I am basically asking why is it true?
3. in "model2.py" why are you using  g = self.dense(x)? it is also a layer I haven't encountered in the paper. I see that it does convolution with the same number of input and output layers. what does it mean?

Thanks in advance for your help!

**Comment by SaoYan** [comment]
1. I found that softmax tends to surpass most "pixels" of the attention map and activate only a small number of pixels.
2. Using adaptive_avg_pool2d implements global average pooling. Please read the PyTorch API documents about the arguments.
3. self.dense is the FC-1 in the paper.
